migrator/src/main/java/com/google/phonenumbers/migrator/MigrationJob.java

```diff
@@ -0,0 +1,112 @@
+/*
+ * Copyright (C) 2020 The Libphonenumber Authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.google.phonenumbers.migrator;
+
+import com.google.common.collect.ImmutableMap;
+import com.google.i18n.phonenumbers.metadata.RangeSpecification;
+import com.google.i18n.phonenumbers.metadata.RangeTree;
+import com.google.i18n.phonenumbers.metadata.i18n.PhoneRegion;
+import com.google.i18n.phonenumbers.metadata.table.CsvTable;
+import com.google.i18n.phonenumbers.metadata.table.RangeKey;
+import java.io.FileReader;
+import java.io.IOException;
+import java.io.InputStreamReader;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Collection;
+import java.util.List;
+import java.util.Map;
+import java.util.Scanner;
+
+/**
+ * Represents a migration operation for a given region where each {@link MigrationJob} contains
+ * a map of E.164 numbers to be migrated as well as the {@link CsvTable} which will
+ * hold the available recipes that can be performed on the range. The number range map is a key value
+ * pair of the E.164 {@link RangeSpecification} representation of a number along with the raw input
+ * String originally entered. Only recipes from the given two digit BCP-47 regionCode will be used.
+ */
+public final class MigrationJob {
+
+  private final CsvTable<RangeKey> recipesTable;
+  private final ImmutableMap<RangeSpecification, String> numberRangeMap;
+  private final PhoneRegion regionCode;
+
+  public MigrationJob(ImmutableMap<RangeSpecification,
```

Comment:
Having a factory to create instances *and* a public constructor could be confusing. At the very least this can be made package-private, but let's chat about static factories vs static factory methods soon.